pkg/controller/storagecluster/reconcile.go

```diff
@@ -863,6 +881,22 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster) []rook.StorageClassDevi
 	return storageClassDeviceSets
 }
 
+func (r *ReconcileStorageCluster) isStorageClassProvisionerType(storageClassname, provisioner string, parameterTypes []string) (bool, error) {
+	storageClass := &storagev1.StorageClass{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: storageClassname}, storageClass)
+	if err != nil {
+		return false, fmt.Errorf("failed to retrieve Storage Class %q. %+v", storageClassname, err)
+	}
+	if storageClass.Provisioner == provisioner {
```

Comment:
Instead of an `if`, use a `select` statement with one `case` for aws-ebs. Specify the throttled disk types in a const and reference them there.